client/src/pages/Profile/Profile.tsx

```diff
@@ -97,14 +100,19 @@ export default function Profile(): JSX.Element {
 
   const submitNewPic = async (e: ChangeEvent<HTMLInputElement>) => {
     if (e.target.files) {
-      setSelectedFile(e.target.files[0]);
+      setLoading(true);
       try {
         const formData = new FormData();
         formData.append('image', e.target.files[0], e.target.files[0].name);
         const newPic = await updateProfile(formData);
         setNewProfilePic(newPic);
+        setLoading(false);
+        const { success } = await loginWithCookies();
```

Comment:
This will work but in the future you may want to consider refactoring the upload image controller so that you can avoid this extra API call :)

Reply:
I thought about the same thing, but then I would have to route through the same stuff on the back end and this was sooo much easier haha. I can go back and change it now or we can save that for later if we have higher priority items. You tell me. :)